internal/kafka/internal/handlers/service_accounts.go

```diff
@@ -124,11 +130,17 @@ func (s serviceAccountsHandler) ResetServiceAccountCredential(w http.ResponseWri
 
 func (s serviceAccountsHandler) GetServiceAccountByClientId(w http.ResponseWriter, r *http.Request) {
 	clientId := r.FormValue("client_id")
+	validators := []handlers.Validate{
+		handlers.ValidateLength(&clientId, "client_id", handlers.MinRequiredFieldLength, &handlers.MaxServiceAccountClientId),
+	}
+
+	if s.service.GetConfig().SelectSSOProvider == keycloak.MAS_SSO {
+		// only service accounts from mas sso are prefixed with "srvc-acc-"
+		validators = append(validators, handlers.ValidateServiceAccountClientId(&clientId, "client_id"))
+	}
+
```

Comment:
I think it would be more readable if, instead of this change, we would add the `SelectSSOProvider` as parameter to `ValidateServiceAccountClientId` which, in turn, will just return `true` if the provider is not `MAS_SSO`.
WDYT?

Reply:
makes sense, i'll change that

Reply:
updated, had to pass in the `SelectSSOProvider` as param to the validator